app/components/utils/mandatee-selector.js

```diff
@@ -33,26 +40,32 @@ export default class MandateeSelector extends Component {
       yield this.initialLoad;
     }
     const queryOptions = {
-      ...this.defaultQueryOptions,  // clone
+      ...this.defaultQueryOptions, // clone
     };
     if (searchTerm) {
       queryOptions['filter[person][last-name]'] = searchTerm;
     }
-    queryOptions['filter[government-body][:uri:]'] = CURRENT_GOVERNMENT_BODY;
+    queryOptions['filter[:lte:start]'] =
+      this.governmentBodyOfDate.toISOString();
+    queryOptions['filter[:gte:end]'] =
+      this.governmentBodyOfDate.toISOString();
+
+    let results = yield this.store.query('mandatee', queryOptions);
 
-    const results = yield this.store.query('mandatee', queryOptions);
-    // Many versions of a mandatee exist within a government-body.
-    // We only want the mandatees with no end-date or an end-date in the future.
-    // mu-cl-resources doesn't have :has-no:-capability for properties.
-    return results.filter((mandatee) => {
-      return !mandatee.end || (mandatee.end > new Date());
-    });
+    if (results.length === 0) {
+      results = yield this.loadCurrentBodyMandatees(searchTerm);
+    }
+    return results;
```

Comment:
The condition of this fallback is weird. Results may for example not be found because the user is searching for a mandatee name that wasn't active in the government at that time. It's unexpected that we start hinting currently active mandatees in that case.

To get the expected behaviour, a query needs to be done before the mandatee-query to get the current active government-body (regeringssamenstelling) based on the date passed as argument. The URI of that government-body should be added to the next filter for mandatees (as was done in the previous implementation with the hardcoded `CURRENT_GOVERNMENT_BODY`).

As stated in the comment that has been removed `mu-cl-resources doesn't have :has-no:-capability for properties.` That filter flag is [only applicable for relationships](https://github.com/mu-semtech/mu-cl-resources#special-filters). The filtering on end-date should therefore be done in the frontend like before.

To query for the goverment-body at the right level (regeringssamenstelling) it's probably worth reading [the data model documentation about the different levels](https://themis.vlaanderen.be/docs/catalogs#ministers).

Reply:
About `:has-no:`: there's a whole thread in the #mu-semtech channel on this. (NHtxEJ9j5SjeBnwvW)

Basically this is translated to `FILTER(NOT EXISTS { ?mandataris mandaat:einde ?mandaatEinde })` instead of `FILTER(!BOUND(?mandaatEinde))`. But I think the two are equivalent, no?

In any case, these two queries return the same result, so I'm quite sure this'll keep on working:

```
PREFIX mandaat: <http://data.vlaanderen.be/ns/mandaat#>
PREFIX prov: <http://www.w3.org/ns/prov#>
PREFIX besluit: <http://data.vlaanderen.be/ns/besluit#>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT ?bestuursOrgaan ?mandataris ?mandaatStart ?mandaatEinde WHERE {
  ?bestuursOrgaan a besluit:Bestuursorgaan;
	prov:hadMember ?mandataris .
  ?mandataris mandaat:start ?mandaatStart .
  OPTIONAL { ?mandataris mandaat:einde ?mandaatEinde . }
  FILTER (!BOUND(?mandaatEinde))
}
```

and

```
PREFIX mandaat: <http://data.vlaanderen.be/ns/mandaat#>
PREFIX prov: <http://www.w3.org/ns/prov#>
PREFIX besluit: <http://data.vlaanderen.be/ns/besluit#>
PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>

SELECT DISTINCT ?bestuursOrgaan ?mandataris ?mandaatStart ?mandaatEinde WHERE {
  ?bestuursOrgaan a besluit:Bestuursorgaan;
	prov:hadMember ?mandataris .
  ?mandataris mandaat:start ?mandaatStart .
  OPTIONAL { ?mandataris mandaat:einde ?mandaatEinde . }
  FILTER (NOT EXISTS { ?mandataris mandaat:einde  ?mandaatEinde })
}
```

Given how much of a refactor/hassle this is to implement without `: has-no:`, I think it's worth the 'risk' here.

Reply:
> The condition of this fallback is weird. Results may for example not be found because the user is searching for a mandatee name that wasn't active in the government at that time. It's unexpected that we start hinting currently active mandatees in that case.

This is true. I would suggest to do this without constants though, by making the call to get the current government body first, using the current Date. Then we can check whether the reference date is earlier or later than the start date of the mandatees in that government composition.

If earlier: do the call using the reference date.
If later: use the current government body we just got.

This might even be more efficient, since the current composition will be used in 99% of the cases.

Reply:
As @madnificent mentioned in the referenced chat discussion there is a risk that `:has-no:` to filter on properties will break randomly in a minor release since it's unspecified behaviour. Because the filtering logic in mu-cl-resources is actively being developed on at the moment, I do see a substantial risk. If we decide to take that risk, we should at least document it inline and provide a test case that fails if the filter doesn't work anymore such that the breaking change becomes apparent.

About the logic to get the mandatees, the logical steps in my head are:
1. Get the government body based on the reference date  (or current date if none is passed)
2. Get the mandatees filtered by:
    - the government body from step 1
    - mandate start date < reference date
    - mandate end date > reference date or undefined (preferably in frontend)
    - roles

We're missing mandatees not linked to a government composition in that case, but I'm wondering whether those should be selectable?  When checking production data using the SPARQL query below, it only yields non-Themis mandatees as a result. Aren't these planned to be phased out/mapped to a Themis mandatee? To ensure data quality it seems acceptable to me that they cannot select those mandatees in the dropdown. Any opinion on this @MikiDi ? 
```sparql
SELECT DISTINCT ?s WHERE {
    ?s a <http://data.vlaanderen.be/ns/mandaat#Mandataris>  .    
    FILTER NOT EXISTS { ?b <http://www.w3.org/ns/prov#hadMember> ?s }
}
```